api/python/t4/packages.py

```diff
@@ -452,15 +459,25 @@ def fetch(self, dest='./'):
         """
         nice_dest = fix_url(dest).rstrip('/')
         file_list = []
-        
+        new_physical_keys = []
+
         for logical_key, entry in self.walk():
             logical_key = quote(logical_key)
             physical_key = _to_singleton(entry.physical_keys)
             new_physical_key = f'{nice_dest}/{logical_key}'
+            new_physical_keys.append(new_physical_key)
             file_list.append((physical_key, new_physical_key, entry.size, entry.meta))
 
         copy_file_list(file_list)
 
+        # return a package reroot package physical keys after the copy operation succeeds
+        # see GH#388 for context
+        pkg = Package()
+        for (logical_key, entry), physical_key in zip(self.walk(), new_physical_keys):
+            pkg.set(logical_key, physical_key)
```

Comment:
So I'm still worried about the performance hit here - especially since it's a regession.

Here's an idea: in the first `for` loop, do something like:
```
new_entry = entry._clone()
new_entry.physical_keys = [new_physical_key]
pkg.set(logical_key, new_entry)
```

That will just accept the new entry and skip checking the size, meta, etc.

Reply:
Done.